### PR TITLE
Done testing for the enumerables functions

### PR DESCRIPTION
**Pull request by Oghenebrume50**

### CODE REVIEW

File: spec/enumerable_spec.rb [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    it "should return an enumerator" do
+      expect(array.my_each_with_index).to be_an_instance_of(Enumerator)
+    end
+
+    it "should return same Array as array" do
+      expect(array.my_each_with_index {|values, index| " #{index} "}).to be_an_instance_of(Array)
+    end
+
+    it "Should return same array as input (but output array  indexes)" do
+      expect(array.my_each_with_index {|values, index| puts index}).to eq([10, 11, 12])
```
Comment (inhaq) [inline spec/enumerable_spec.rb]: no need to use `puts`

File: spec/enumerable_spec.rb [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  end
+
+  describe "#my_each_with_index" do
+    let (:array) {[10, 11, 12]}
+    it "should return an enumerator" do
+      expect(array.my_each_with_index).to be_an_instance_of(Enumerator)
+    end
+
+    it "should return same Array as array" do
+      expect(array.my_each_with_index {|values, index| " #{index} "}).to be_an_instance_of(Array)
```
Comment (inhaq) [inline spec/enumerable_spec.rb]: no need to use interpolation

File: spec/enumerable_spec.rb [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    end
+
+    it "Should return same array as input (but output array  indexes)" do
+      expect(array.my_each_with_index {|values, index| puts index}).to eq([10, 11, 12])
+    end
+
+    it "Should return same array as input (but output array Values)" do
+      expect(array.my_each_with_index {|values, index| puts values}).to eq([10, 11, 12])
```
Comment (inhaq) [inline spec/enumerable_spec.rb]: no need to use `puts`

File: spec/enumerable_spec.rb [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+      expect(array.my_each).to be_an_instance_of(Enumerator)
+    end
+
+    it "should return same Array as array" do
+      expect(array.my_each {|x| " #{x} "}).to be_an_instance_of(Array)
+    end
+  end
+
+  describe "#my_all" do
+    let (:array) {[1,2,3,4,5]}
```
Comment (inhaq) [inline spec/enumerable_spec.rb]: `DRY` use let once at the top

**Comment by Oghenebrume50** [comment]
@inhaq  done requested changes, I used let twice, once on top, but another let was used to represent a different array

File: spec/enumerable_spec.rb (hunk 3 above)
Comment (shloch) [reply spec/enumerable_spec.rb]: all puts removed , thanks

**Pull request merged**